src/libs/swapAndBridge/swapAndBridge.ts

```diff
@@ -24,6 +24,54 @@ import { PaymasterService } from '../erc7677/types'
 import { TokenResult } from '../portfolio'
 import { getTokenBalanceInUSD } from '../portfolio/helpers'
 
+/**
+ * Put a list of all the banned addresses (left side) with their
+ * corresponding valid/correct address (right side).
+ * This is fixing the case of token duplication like EURe. / CELO
+ * where if the incorrect (outdated) address is used instead of the
+ * new one in the swap, no routes will be found
+ */
+const getBannedToValidAddresses = (): {
+  [chainId: string]: { [bannedAddr: string]: string }
+} => {
+  /** ****************************************************
+   *        MAKE SURE ADDRESSES ARE CHECKSUMMED
+   ****************************************************** */
+  const bannedEurePolygon = '0xE0aEa583266584DafBB3f9C3211d5588c73fEa8d'
+  const validEurePolygon = '0x18ec0A6E18E5bc3784fDd3a3634b31245ab704F6'
+
+  /** ****************************************************
+   *        MAKE SURE ADDRESSES ARE CHECKSUMMED
+   ****************************************************** */
+  const bannedEureGnosis = '0x420CA0f9B9b604cE0fd9C18EF134C705e5Fa3430'
+  const validEureGnosis = '0xcB444e90D8198415266c6a2724b7900fb12FC56E'
+
+  /** ****************************************************
+   *        MAKE SURE ADDRESSES ARE CHECKSUMMED
+   ****************************************************** */
+  const bannedCelo = '0x471EcE3750Da237f93B8E339c536989b8978a438'
+  const validCelo = ZeroAddress
+
+  return {
+    '137': {
+      [bannedEurePolygon]: validEurePolygon
+    },
+    '100': {
+      [bannedEureGnosis]: validEureGnosis
+    },
+    '42220': {
+      [bannedCelo]: validCelo
+    }
+  }
+}
```

Comment:
Let's discuss this approach offline. For the EURe case, I wouldn't call these "banned". They are v1 and v2 token contracts. The one marked as "old" in coingecko are actually the ones we need to use (v1) to get quotes (because v2 token contracts are not yet supported widely). Ref: https://monerium.dev/docs/contracts-v2

Also, this on theory is temporary patch. We need to monitor the situation. Maybe cross-check with LiFi they've thought for a solution to this case.